File: pys/inthenews.py
import requests
from bs4 import BeautifulSoup

url = 'http://inthenews.co.kr/latest-news/'
req = requests.get(url)
bs = BeautifulSoup(req.text, 'lxml')

wrapper = bs.find('div', class_='pt-cv-wrapper')
items = wrapper.find_all('div', class_='pt-cv-ifield')

for item in items[:4]:
    h4_tag = item.find('h4', class_='pt-cv-title')
    a_tag = h4_tag.find('a')
    href = a_tag['href']
    title = a_tag.get_text().strip()

    req2 = requests.get(href)
    bs2 = BeautifulSoup(req2.text, 'lxml')

    date = bs2.find('div', class_='post-date').get_text().strip()

    wrapper2 = bs2.find('div', class_='pf-content')
    contents = ''.join([it.get_text().strip() for it in wrapper2.find_all('p')])

    print(title, href, date, contents)

File: crawlers/inthenews.go
package crawlers

import (
	"fmt"
	"net/http"

	"flashnews/models"
	"flashnews/utils"

	"github.com/PuerkitoBio/goquery"
)

const inTheNewsCommonURL = "http://inthenews.co.kr/latest-news/"
const inTheNewsItemURL = ""

type InTheNews struct{}

func (c InTheNews) GetName() string {
	return "inthenews"
}

func (c InTheNews) GetGroup() string {
	return "4"
}

func (c InTheNews) GetList(number int) ([]models.NewsItem, error) {
	// Number
	var _number int
	if number > 10 || number < 1 {
		_number = 10
	} else {
		_number = number
	}
	result := make([]models.NewsItem, _number)

	// Request
	req, err := http.Get(inTheNewsCommonURL)
	if err != nil {
		return result, err
	}
	defer req.Body.Close()

	if req.StatusCode != 200 {
		return result, fmt.Errorf("[ERROR] Request Status Code : %d, %s", req.StatusCode, req.Status)
	}

	// HTML Load
	html, err := goquery.NewDocumentFromReader(req.Body)
	if err != nil {
		return result, err
	}

	// Parsing
	wrapper := html.Find("div.pt-cv-wrapper")
	items := wrapper.Find("div.pt-cv-ifield")
	items.Each(func(i int, sel *goquery.Selection) {
		if i >= _number {
			return
		}

		h4Tag := sel.Find("h4.pt-cv-title")
		aTag := h4Tag.Find("a")
		href, ok := aTag.Attr("href")
		if !ok {
			result[i] = models.NewsItem{}
			return
		}
		url := inTheNewsItemURL + href

		date := ""
		title := aTag.Text()

		result[i] = models.NewsItem{
			Title:    title,
			URL:      url,
			Contents: "",
			Datetime: date,
		}
	})

	return result, nil
}

func (c InTheNews) GetContents(item *models.NewsItem) error {
	// Request
	req, err := http.Get(item.URL)
	if err != nil {
		return err
	}
	defer req.Body.Close()

	if req.StatusCode != 200 {
		return fmt.Errorf("[ERROR] Request Status Code : %d, %s", req.StatusCode, req.Status)
	}

	// HTML Load
	html, err := goquery.NewDocumentFromReader(req.Body)
	if err != nil {
		return err
	}

	// Parsing
	date := utils.TrimAll(html.Find("div.post-date").Text())
	item.Datetime = date

	wrapper := html.Find("div.pf-content")
	contents := ""
	wrapper.Find("p").Each(func(idx int, sel *goquery.Selection) {
		contents += sel.Text()
	})
	contents = utils.TrimAll(contents)

	item.Contents = contents
	return nil
}
